addons/nuxeo-binary-metadata/src/main/java/org/nuxeo/binary/metadata/internals/MetadataMappingDescriptor.java

```diff
@@ -45,6 +46,10 @@
 
     @XNode("@ignorePrefix")
     protected Boolean ignorePrefix = true;
+    
+    /** @since 11.1 */
+    @XNode("@readOnly")
+    protected Boolean readOnly = false;
```

Comment:
Using default values in the fields makes it impossible to have two contributions merging each other without redefining a field (I know there's no merge right now but it's to prepare for the future). You should just have `protected Boolean readOnly` (so default to `null`).

Same for `ignorePrefix` except that because you want the default to be true, make the getter be `return !Boolean.FALSE.equals(ignorePrefix)`. You can check that the end result is the same.